install/common-pre-install/hpcm_installation-cpi.md

```diff
@@ -0,0 +1,1076 @@
+               ```
+
+               Perform switch discovery.
+
+               ```bash
+               cm node add -c mswitch.conf
+               ```
+
+         1. For PDU : Create the PDU configuration file. Here is the example of a PSU in configuration file format.
+
+               ```text
+               [discover]
+               internal_name=pdu-x3000-001, mgmt_bmc_net_name=head-bmc, geolocation="cold isle 4 rack 1 B power",mgmt_bmc_net_macs=00:0a:9c:62:04:ee,hostname1=pdu-x3000-001, pdu_protocol="snmp/admn"
+               ```
+
+               Perform PDU discovery.
+
+               ```bash
+               cm node add -c pdus.conf
+               ```
+
+         1. For Fabric Switch: Use cm controller add to perform discovery. Add only River cabinent fabric switches
+
+            ```bash
+            cm controller add -c sw-hsn-x3000-001 -t external_switch -m 00:40:a6:82:f7:5f -u root -p initial0
+            ```
+
+         1. For CMC : Use cm controller add to perform discovery.
+
+            For gigabyte server.
+
+            ```bash
+            cm controller add -c SubRack001-cmc -t gigabyte -m  b4:2e:99:b8:da:03 -u root -p initial0
+            ```
+
+         1. For CEC : TBD.
+
+      1. Auto-discovery method.
+
+         Management, worker, storage nodes, UANs will be auto discovered and the procedure is described as follows -
+
+         >__Note:__
+         >
+         > - Ensure all the river components are powered on, dhcp is enabled on bmc and only one switch out of the available spine switch.
+         >
+         > - If there are multiple mellanox spine switches then there should be only 1 spine switch with active port connection to NCNs. On other switches, connections to NCNs should be disabled till all nodes have been booted with HPCM Images.
+         >
+         >  ```bash
+         >  ssh admin@<ip-addr>
+         >  enable
+         >  configure terminal
+         >  interface ethernet 1/2-1/9
+         >  shutdown
+         >  ```
+         >
+         > Once all the NCN's are booted enable all the ports by running `no shutdown` command.
+
+         1. Enable auto-discovery process.
+
+            The management nodes, worker nodes, storage nodes, UANs and CNs can be auto discovered using the following command:
+
+            ```bash
+            cm node discover enable 
+            ```
+
+            >__Note:__ If there are any intel nodes run the following command:
+            >
+            >```bash
+            >sed -i '226s/ipmitool lan print/ipmitool lan print 3/g' /opt/clmgr/tools/cm_pxe_status
+            >```
+
+         1. Check the power status and pxe status.
+
+            > __Note:__ Verify if all the river node BMCs are leased with an IP.
+
+            Check the power status of nodes.
+
+            ```bash
+            cm node discover status | grep BMC  > temp.txt

```

Comment:
line 671 needs to be changed as below;
cm node discover status | grep client-hostname  > temp.txt

common change: remove username and psswrd from all places